### DISCUSSION TITLE
Option to Suppress Error Messages

### DISCUSSION THREAD

**Discussion by SirBenJammin** [dbeaver/dbeaver#190] [Q&A] [answered]
Hey Serge,

Is it possible that (if possible certain) errors messages can be suppressed to the error log only. Prime example is the use of two db connections (for the same db) when connecting at two different locations (i.e. I have one db connection for the office and another when working from home via VPN).

It can be rather annoying when each time I have to change the SQL scripts db connection, I have to handle the expected error message (i.e. Unable to connect/Login timeout for Connection).

Thanks Serge for producing a fantastic piece of software, it has help me massively in my day to day workload. Good job :+1: :)

Thanks,

Ben

**Comment by serge-rider** [comment]
Could you clarify it a bit.
Do you need to suppress all error message boxes or just some specific ones?

**Comment by SirBenJammin** [comment]
The error suppression is mainly for the transfer of one db connection to another, for example when going from a standard IP connection (office) to a VPN IP connection (remote) per each individual SQL script. When there are over 20 SQL scripts to transfer the constant notification that the connection has failed (office) getting rather tedious.

**Comment by serge-rider** [comment]
I'd say it is a pretty specific environment.
Did you try to change connection for all your scripts just in one click in scrip browser? See below.
![assoc-scripts](https://cloud.githubusercontent.com/assets/6398845/12974157/d2d0def8-d0c2-11e5-8ae3-e7b014ec8fc5.png)
This re-association won't throw any errors at all.
If this approach doesn't look good by any reason - please describe these reasons.

**Comment by SirBenJammin** [comment]
Interesting... I was already aware of the re-association within the script browser. Using this process requires to constantly close all open SQL scripts (which can be 20+ at any one time), each time I move location and use a VPN.

I believe the re-connection process should be much simpler, there should be a option to just report connection errors to the error log, rather than a modal error dialog. The reason is I know that the connection is not going to work (as it requires a different IP address to connect to the same db), I don't need a constant reminder each time I need to re-connect using a different connection.

The proposed process would be to close all open SQL scripts, go into the script browser, try and find the closed scripts (within 1000's of numbered scripts like in your example) and re-associate the data source once I am in the new location using the new connection and then re open related scripts. This process would have happen everyday for me.

**Comment by serge-rider** [comment, marked as answer]
Generally this looks like network configuration issues workaround.
If the only difference between work and home environment than possible workaround is to add host alias. In /etc/hosts (C:\Windows\System32\drivers\etc) add the same alias with different IPs. So you won't need to change anything in DBeaver at all. Isn't this a solution?

**Comment by SirBenJammin** [comment]
Wow I completely forgot about the host file. This is a solution, thanks Serge.

**Comment by serge-rider** [comment]
Ok, let's assume this is the default workaround.
